tests/dummy.c

```diff
@@ -133,4 +140,16 @@ int socket__dummy(struct __sk_buff *skb)
 	return 0;
 }
 
+SEC("sk/skb/parser/dummy_sockmap")
+int parser_dummy(struct __sk_buff *skb)
```

Comment:
The parameter should be `struct sk_msg_md *msg`.

Ditto in `verdict_dummy`.

Reply:
I think `struct __sk_buff *` is correct:
- https://github.com/torvalds/linux/blob/master/tools/testing/selftests/bpf/progs/sockmap_parse_prog.c#L16
- https://elixir.bootlin.com/linux/v4.18/source/kernel/bpf/sockmap.c#L1497

and the verifier converts `struct __sk_buff *` to the real `struct sk_buff *` at load time. Maybe I'm missing something?